File: src/main/java/com/aquatrack/service/BuildingService.java
package com.aquatrack.service;

import com.aquatrack.dto.building.BuildingRequestDto;
import com.aquatrack.dto.building.BuildingResponseDto;

import java.util.List;

/**
 * Service interface for Building management.
 */
public interface BuildingService {

    /**
     * Creates a new building for an apartment.
     *
     * @param apartmentId Apartment ID
     * @param requestDto Building request data
     * @return Created building details
     */
    BuildingResponseDto createBuilding(Long apartmentId,
                                       BuildingRequestDto requestDto);

    /**
     * Returns all buildings of an apartment.
     *
     * @param apartmentId Apartment ID
     * @return List of buildings
     */
    List<BuildingResponseDto> getAllBuildings(Long apartmentId);

    /**
     * Returns building details.
     *
     * @param apartmentId Apartment ID
     * @param buildingId Building ID
     * @return Building details
     */
    BuildingResponseDto getBuildingById(Long apartmentId,
                                        Long buildingId);

    /**
     * Updates building details.
     *
     * @param apartmentId Apartment ID
     * @param buildingId Building ID
     * @param requestDto Updated building data
     * @return Updated building details
     */
    BuildingResponseDto updateBuilding(Long apartmentId,
                                       Long buildingId,
                                       BuildingRequestDto requestDto);

    /**
     * Deletes a building.
     *
     * @param apartmentId Apartment ID
     * @param buildingId Building ID
     */
    void deleteBuilding(Long apartmentId,
                        Long buildingId);

    /**
     * Returns all buildings assigned to the logged-in manager.
     *
     * @return List of assigned buildings
     */
    List<BuildingResponseDto> getManagerBuildings();

}

File: Aquatrack-frontend/src/services/buildingService.js
import api from "../api/axios";

// ==========================================
// Get Buildings By Apartment
// ==========================================

export async function getBuildings(apartmentId) {

    const response = await api.get(

        `/apartments/${apartmentId}/buildings`

    );

    return response.data;

}

// ==========================================
// Create Building
// ==========================================

export async function createBuilding(
    apartmentId,
    data
) {

    const response = await api.post(

        `/apartments/${apartmentId}/buildings`,

        data

    );

    return response.data;

}

// ==========================================
// Get Building
// ==========================================

export async function getBuilding(
    apartmentId,
    buildingId
) {

    const response = await api.get(

        `/apartments/${apartmentId}/buildings/${buildingId}`

    );

    return response.data;

}

// ==============================
// Manager Buildings
// ==============================

export const getManagerBuildings = async () => {

    const response = await api.get(
        "/manager/buildings"
    );

    return response.data;

};

// ==========================================
// Update Building
// ==========================================

export async function updateBuilding(
    apartmentId,
    buildingId,
    data
) {

    const response = await api.put(

        `/apartments/${apartmentId}/buildings/${buildingId}`,

        data

    );

    return response.data;

}

// ==========================================
// Delete Building
// ==========================================

export async function deleteBuilding(
    apartmentId,
    buildingId
) {

    const response = await api.delete(

        `/apartments/${apartmentId}/buildings/${buildingId}`

    );

    return response.data;

}